### DISCUSSION TITLE
what parameter is making the loss progress crazy in a fine-tuning fit

### DISCUSSION THREAD

**Discussion by bernstei** [ACEsuit/mace#1455] [Q&A] [answered]
I have two fits on the same dataset, with slightly different parameters, and in once case the loss goes smoothly downward, and in the other it drops for 1-2 epochs, then ~~jumps up~~ starts increasing for about 20 epochs, and then slowly goes down but never as low as epoch 2.  Any ideas which of these parameters are the relevant ones to affect this?   

My first thought was lr, but both claim to be using lr of 0.0001, in one case explicit (with override_lr), and in the other because I set it to 0.001 but forgot to override, so it's using the default 0.0001 because multihead replay is active.

The only other ones that seem like they might be important on that scale are the loss itself, and clip_grad. I'm testing, but curious to see what people's intuition suggests.

```
100c100
< foundation_model_elements=True,
---
> foundation_model_elements=False,
118,119c118,119
< loss='universal',
< forces_weight=10.0,
---
> loss='weighted',
> forces_weight=1.0,
121c121
< energy_weight=10.0,
---
> energy_weight=1.0,
137c137
< batch_size=8,
---
> batch_size=10,
141c141
< weight_decay=0.0,
---
> weight_decay=5e-07,
156,157c156,157
< ema_decay=0.9999,
< max_num_epochs=200,
---
> ema_decay=0.995,
> max_num_epochs=150,
167c167
< clip_grad=1.0,
---
> clip_grad=100.0,
```

**Comment by bernstei** [comment]
Looks like it's mainly the `energy_weight` and `force_weight` being 10 instead of 1. Adding that to my setup, which is basically the `mace_run_Train` default, is sufficient to recover smoother decrease of the loss.  The differences in other settings may have a small effect, but that's the main one.  Note that decreasing `weight_pt_head` from 1.0 to 0.1 _does not_ behave the same.  The absolute magnitude of the energy/force weight is critical. 

This seems bad to me.

**Reply by bernstei** [reply, marked as answer]
Looks like with default `weight_decay = 5e-7`, the behavior is quite sensitive to the magnitude of the force and energy weights.  with `weight_decay = 0`, which is recommended for fine-tuning anyway, it's much less sensitive

**Comment by ilyes319** [comment]
For future reference here, I recommend people to use `weight_decay = 0`. The weight decay introduces an intrinsic scale to the loss when using adam (not adamW) see here: https://arxiv.org/abs/1711.05101. Note that setting non zero weight_decay is not recommened for is not recommended for finetuning.
